modules/vertice-cor-report/main.tf

```diff
@@ -0,0 +1,45 @@
+locals {
+  defaults_columns_for_selection = ["account_id", "action_type", "currency_code", "current_resource_details", "current_resource_summary",
+    "current_resource_type", "estimated_monthly_cost_after_discount", "estimated_monthly_cost_before_discount",
+    "estimated_monthly_savings_after_discount", "estimated_monthly_savings_before_discount",
+    "estimated_savings_percentage_after_discount", "estimated_savings_percentage_before_discount",
+    "implementation_effort", "last_refresh_timestamp", "recommendation_id", "recommendation_lookback_period_in_days",
+    "recommendation_source", "recommended_resource_details", "recommended_resource_summary", "recommended_resource_type",
+  "region", "resource_arn", "restart_needed", "rollback_possible", "tags"]
+  cor_columns_for_selection = length(var.cor_columns_for_selection) == 0 ? join(", ", local.defaults_columns_for_selection) : join(", ", setintersection(local.defaults_columns_for_selection, var.cor_columns_for_selection))
+}
+
+data "aws_s3_bucket" "vertice_cor_bucket" {
+  bucket = var.cor_report_bucket_name
+}
+
+resource "aws_bcmdataexports_export" "vertice_cor_report" {
```

Comment:
...for example, by adding a `precondition` here:
```hcl
resource "aws_bcmdataexports_export" "vertice_cor_report" {
  precondition {
    condition = setintersection(local.defaults_columns_for_selection, var.cor_columns_for_selection) == var.cor_columns_for_selection
    error_message = "All the columns selected for COR must come from ${local.defaults_columns_for_selection}."
  }
}
```

Or, alternatively, this validation can happen on the variable level (although we might have to duplicate the list of default/allowed values).

Reply:
Ok, that makes sense, I will try to implement a precondition. Thanks for the advice.